cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -566,6 +594,25 @@ TEST_F(ScalarTemporalTest, Strftime) {
                    utf8(), string_microseconds, &options);
   CheckScalarUnary("strftime", timestamp(TimeUnit::NANO, "US/Hawaii"), nanoseconds,
                    utf8(), string_nanoseconds, &options);
+
+  auto options_hms = StrftimeOptions("%H:%M:%S");
+  const char* times_s = R"([59, null])";
+  const char* times_ms = R"([59123, null])";
+  const char* times_us = R"([59123456, null])";
+  const char* times_ns = R"([59123456789, null])";
+  const char* strftime_ns = R"(["00:00:59.123456789", null])";
+  const char* strftime_us = R"(["00:00:59.123456", null])";
+  const char* strftime_ms = R"(["00:00:59.123", null])";
+  const char* strftime_s = R"(["00:00:59", null])";
+
+  CheckScalarUnary("strftime", time64(TimeUnit::NANO), times_ns, utf8(), strftime_ns,
+                   &options_hms);
+  CheckScalarUnary("strftime", time64(TimeUnit::MICRO), times_us, utf8(), strftime_us,
+                   &options_hms);
+  CheckScalarUnary("strftime", time32(TimeUnit::MILLI), times_ms, utf8(), strftime_ms,
+                   &options_hms);
+  CheckScalarUnary("strftime", time32(TimeUnit::SECOND), times_s, utf8(), strftime_s,
```

Comment:
Should we 1) check what happens for fields like year, and 2) check that `%z` isn't allowed for times below in StrftimeNoTimezone?

Reply:
Added.